ls1intum/Artemis
# `General`: Prevent duplicated alerts in some places

### Checklist
#### General


#### Client


### Motivation and Context



Due to the new alerts service that unified error and regular alerts into one module, some cases exist where an error spawned two alerts. This is due to the global error handler spawning alerts even if it doesn't make sense.

### Description

- Rework a few default cases and conditions in the http error handler to only spawn alerts if the error returned by the server really contains a meaningful error. 
- Disabled default alerts for 404s as they didn't contain any meaningful information for a user.
- Added the "skipAlert" flag to grading update endpoint exception as the component spawns alerts already

### Steps for Testing

Examples:
- Go to Management View of a Programming Exercise > Grading. Try to set the weight of a test case to -1. It should only spawn one alert now.
- Try to create a programming exercise with a title that already exists. It should show a meaningful error message (in English; translation was not properly done  here when this was introduced and is out of scope for the PR) instead of a key

### Future work

In the future we need to consolidate all of the error handling and alert spawning.
Alerts spawned by the interceptor usually display the default message from the server because translation was not done properly. We need to find a way to unify all error handling from HTTP.
Whereever error handlers handle 400 errors within the component, the "skipAlert" flag on the server side exception should be set like it was exemplary done for grading updates.
This is especially important if the app is running in german mode, as the duplicate alert prevention doesn't properly work then.

Ideally we rework all errorKeys sent from the server side to actually make sense, define proper translations for them, and provide a default handler that can be used whereever a request is made. On each code location in the client where a request is made, we then need to decide if we want to use the default error handler or use a special error handling procedure.

For errors >= 500, we could keep a default http interceptor as those are usually unexpected and can't or shouldn't be prepared or expected in component / service code

### Review Progress





#### Code Review
#### Manual Tests


### Screenshots



Before:
<img width="813" alt="grafik" src="https://user-images.githubusercontent.com/23171488/156815262-77eef20b-b9db-4701-9948-5ecdba49c543.png">
After: The second alert is not shown
